src/transformStateWithClones.js

```diff
@@ -7,7 +7,34 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  let currentState = { ...state }; // Create a clone of the initial state
+
+  for (const action of actions) {
+    const { type } = action;
+
+    if (type === 'addProperties') {
```

Comment:
Use destructuring assignment for the action properties. Example: const { type, extraData } = action;